ext/background/background.js

```diff
@@ -442,7 +452,7 @@ async function providerInfo(url, title, force) {
     provider_name: `${domain}.${tld} (via Range Sync)`,
     // prefixing 'chromeext_' will make it easier to find out what providers
     // to add next
-    source_id: `chromeext_${base}`,
+    source_id: `chromeext_${base + search}`,
```

Comment:
Are we ok with the character set here?

Reply:
What do you propose? Hash it to keep it limited without compromising uniqueness?

Reply:
I'd need to check the code paths about whether there's any assumptions about `source_id`. I think it's probably safer as simple alpha-numeric character set, and having the length be unbounded by also be a problem.

Hashing is certainly one option.

We have this hash algorithm in the client code:

```js
// Implementation of Java's String.hashCode. Not secure.
exports.hashCode = function (str) {
  let hash = 0;
  if (!str || str.length === 0) return hash;
  for (let i = 0; i < str.length; i++) {
    let char = str.charCodeAt(i);
    hash = (hash << 5) - hash + char;
    hash = hash & hash;
  }
  return hash;
};
```

Reply:
Moved hashCode to background.js where it's used https://github.com/range-labs/range-sync/pull/58/commits/0a2ae67452045f1f3fa1f1e15d5d50672662f75f and 1) converted the hash to bytes and 2) encode it with base64 https://github.com/range-labs/range-sync/pull/58/commits/bafb9d3da40ed4b6284bae81d433217c322b338d (easier to read in total diff).

Reply:
The base64 still has `+/`, which doesn't include the `_` we've been using to delimit prefixes.

Reply:
`+/` could be problematic characters. Base64 has variants we could use, but we could likely safely strip them as collisions will be unlikely. 

https://www.base64url.com/

Reply:
I converted it to the url-safe https://github.com/range-labs/range-sync/pull/58/commits/17ad8de5c49e3132dd2359d922e2fd2e5afd1f74.